app/jobs/pdf_generator.rb

```diff
@@ -0,0 +1,103 @@
+require 'json'
+require 'net/http'
+require 'rmagick'
+
+class PdfGenerator
+    @queue = :ingests
+
+    def self.perform(pid)
+      manifest_url = self.config['vufind_test_url'].to_s + "/Item/" + pid + "/Manifest"
+      json = self.do_get manifest_url
+      parsed_json = JSON.parse json
+      large_jpegs = parsed_json["sequences"][0]["canvases"].map do |current|
+        current["images"][0]["resource"]["@id"]
+      end
+      pdf = self.generate_pdf(large_jpegs)
+      self.add_pdf_to_pid(pdf,pid)
+    end
+
+    def self.add_pdf_to_pid(pdf,pid)
+      resource = Fedora3Object.from_pid pid
+      document_list = Fedora3Object.from_next_pid
+      document_list.parent_pid = resource.pid
+      document_list.model_type = "ListCollection"
+      document_list.title = "Document List"
+      document_list.core_ingest("I")
+      document_list.collection_ingest
+      document_list.list_collection_ingest
+      image_data = self.build_document document_list, pdf, 1
+      self.add_datastreams_to_document pdf, image_data
+    end
+
+    def self.build_document(document_list, document, number)
+      document_data = Fedora3Object.from_next_pid
+      document_data.logger = @logger
```

Comment:
I think you can remove this line; we're not using a logger in this code anywhere else, so it's not doing anything.